src/main/logutils.ts

```diff
@@ -129,18 +131,67 @@ const wowProcessStopped = () => {
     isRetailRunning = false;
 
     if (recorder.isRecording) {
-        const videoStopDate = new Date();
-        const milliSeconds = (videoStopDate.getTime() - videoStartDate.getTime()); 
-        metadata.duration = Math.round(milliSeconds / 1000);
-    
-        // Assume loss as game was closed. 
-        metadata.result = false;
-        recorder.stop(metadata);
+        videoStopDate = new Date();
+        endRecording();
     } else if (recorder.isRecordingBuffer) {
         recorder.stopBuffer();
     }
 };
 
+/**
+ * Initiate recording and mark as having something in-progress
+ */
+const startRecording = (category: VideoCategory) => {
+    console.log(`[Logutils] Start recording a video for category: ${category}`)
+
+    // Ensure combatant map and player combatant is clean before
+    // starting a new recording.
+    clearCombatants();
+
+    currentActivity = category;
+    recorder.start();
+};
+
+type EndRecordingOptionsType = {
+    discardVideo?: boolean, // Discard the video/don't save the recording
+    duration?: number,      // Explicitly give video duration
+    result?: boolean,       // Success/Failure result for the overall activity
+};
+
+/**
+ * Stop recording and mark as not doing anything
```

Comment:
Worth stating in comment here that videoStartDate/videoStopDate should be set before calling this.

Reply:
Easy fix - if the method is called without a valid `videoStopDate` we'll set it to `new Date()` because it _needs_ to be set or it'll cause an invalid application state if we bug out of it or throw an error.

Also the whole `duration` thing in `EndRecordingOptionsType` is pointless - no activities specify it anymore. I'll remove it entirely.